Jenkinsfile

```diff
@@ -1,47 +1,99 @@
-  def testStatus = exec(script: command, returnStatus: true)
+def withLtCreds(Closure body) {
+  withCredentials([usernamePassword(credentialsId: 'lambda-engci', usernameVariable: 'LT_USERNAME', passwordVariable: 'LT_ACCESS_KEY')]) {
+    body()
+  }
+}
 
-  echo "Writing JUnit results for ${name} on node: $NODE_NAME"
+def runBedrockTest(String command, Boolean runAll, int retry = 0, int timeout = 0) {
+  def bedrockCmd = command + (runAll ? " --ignore-lerna-changed=true" : "")
+  echo "Running Bedrock cmd: ${command}"
+  def testStatus = sh(script: command, returnStatus: true)
   junit allowEmptyResults: true, testResults: 'scratch/TEST-*.xml'
 
-  // If the tests failed (exit code 4) then just mark it as unstable
   if (testStatus == 4) {
-    unstable("Tests failed for ${name}")
+    unstable("Test failed")
   } else if (testStatus != 0) {
-    error("Unexpected error running tests for ${name} so passing failure as exit code")
+    if (retry > 0) {
+      echo "Running retry [${retry}] after [${timeout}]"
+      sleep(timeout)
+      runBedrockTest(command, runAll, retry - 1, timeout)
+    } else {
+      error("Unexpected error")
+    }
   }
 }
 
-def runBrowserTests(String name, String browser, String os, Integer bucket, Integer buckets, Boolean runAll) {
+def runHeadlessTests(Boolean runAll) {
+  def bedrockCmd = "yarn grunt headless-auto"
+  runBedrockTest(bedrockCmd, runAll)
+}
+
+def runRemoteTests(String name, String browser, String provider, String bucket, String buckets, Boolean runAll, int retry = 0, int timeout = 0) {
+  def awsOpts = " --sishDomain=sish.osu.tiny.work --devicefarmArn=arn:aws:devicefarm:us-west-2:103651136441:testgrid-project:79ff2b40-fe26-440f-9539-53163c25442e"
   def bedrockCommand =
-    "yarn grunt browser-auto" +
-      " --chunk=400" +
-      " --bedrock-os=" + os +
-      " --bedrock-browser=" + browser +
-      " --bucket=" + bucket +
-      " --buckets=" + buckets;
-
-  runTests(name, bedrockCommand, runAll);
+  "yarn browser-test" +
+    " --chunk=400" +
+    " --bedrock-browser=" + browser +
+    " --remote=" + provider +
+    " --bucket=" + bucket +
+    " --buckets=" + buckets +
+    " --name=" + name +
+    "${provider == 'aws' ? awsOpts : ''}"
+    runBedrockTest(bedrockCommand, runAll, retry, timeout)
 }
 
-def runHeadlessTests(Boolean runAll) {
-  def bedrockCommand = "yarn grunt headless-auto";
-  runTests("chrome-headless", bedrockCommand, runAll);
+def runTestPod(String name, String browser, String provider, String bucket, String buckets, Boolean runAll) {
+  return {
+    tinyPods.node([
+      resourceRequestCpu: '6',
+      resourceRequestMemory: '4Gi',
+      resourceLimitCpu: '7.5',
+      resourceLimitMemory: '4Gi'
+    ]) {
+      if (provider == 'aws') {
+        stage('Tunnel') {
+          echo 'Adding Tunnel'
+          sh """
+          echo "sish.osu.tiny.work ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIKl0pmqRdhKQksLYj9k3FUVdrkD/GWXJ/YhHJ3KWRDvz" >> ~/.ssh/known_hosts
```

Comment:
Like wise this do we want this exposed. I guess it's a public key so it doesn't really matter but would be good if it came from a jenkins credentials thing instead easier to change as well.

Reply:
It's already public now (this PR won't disappear), so it doesn't matter much. I agree that we eventually want to move some of these details into `waluigi` to make it easier to support 6 & 7 branches. Is that necessary before this merges though?

Reply:
Yes, I will update to have a secret text instead

Reply:
Moved to secret text instead